src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -195,7 +206,12 @@ public void testWithCacheWithNoViolation() throws Exception {
         final File file = temporaryFolder.newFile("file.java");
         final List<String> lines = new ArrayList<>();
         lines.add(" class a {} ");
-        treeWalker.processFiltered(file, lines);
+        try {
+            treeWalker.processFiltered(file, lines);
+        }
+        catch (CheckstyleException ex) {
+            fail("Exception is not expected");
```

Comment:
@Nimfadora , please remove from this PR all 
```
+        catch (CheckstyleException ex) {
+            fail("exception is not expected");
+        }
```

Please add all cases that are left in suppression of PMD one by one, we will address them one by one, I will give recommendations. Please regenerate your report to contain all suppressed cases - I will provide guidance on by one.